### PR TITLE
rfc: backup format

### PR DESCRIPTION
**Pull request by septs**
closes #

### CODE REVIEW

File: docs/rfc/001-backup-format.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# Structured Backup format
+
+## Principles
+
+- Based-zip archive format
+- Based-zip standard file encryption
```
Comment (Jack-Works) [inline docs/rfc/001-backup-format.md:6]: Is there any JS library that can do these two things easily?

File: docs/rfc/001-backup-format.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Structured Backup format
+
+## Principles
+
+- Based-zip archive format
```
Comment (Jack-Works) [inline docs/rfc/001-backup-format.md:5]: How to handle UTF-8? IMO zip doesn't work well with UTF-8 filename.

File: docs/rfc/001-backup-format.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+  - files = file hash list without manifest* and README.txt
+
+# manifest.json with per persona private key signing
+manifest-[persona-pubkey-hash].signed
+
+permissions.json # original backup file "grantedHostPermissions" field.
+user-groups.json # original backup file "userGroups" field.
+posts.json       # original backup file "posts" field.
+profiles.json    # original backup file "profiles" field.
+personas.json    # original backup file "personas" field.
```
Comment (Jack-Works) [inline docs/rfc/001-backup-format.md:33]: IMO those items should be folders, they might contain binary data.

File: docs/rfc/001-backup-format.md (hunk 2 above)
Comment (septs) [reply docs/rfc/001-backup-format.md:5]: zip64 format

File: docs/rfc/001-backup-format.md (hunk 3 above)
Comment (septs) [reply docs/rfc/001-backup-format.md:33]: ACK

Comment (septs) [reply docs/rfc/001-backup-format.md:33]: i added `resources` subfolder store binary file.

**Comment by septs** [comment]
@Jack-Works the idea, need pre-migrate backup and restore function core code to subpackage.

**Review by Jack-Works — APPROVED** [review]

**Comment by septs** [comment]
@guanbinrui merge ready

File: docs/rfc/001-backup-format.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  - build info
+  - maskbook version
+  - type = maskbook-backup
+  - files = file hash list without manifest* and README.txt
+
+# manifest.json with per persona private key signing
+manifest-[persona-pubkey-hash].signed
+
+permissions.json # original backup file "grantedHostPermissions" field.
+user-groups.json # original backup file "userGroups" field.
```
Comment (Jack-Works) [inline docs/rfc/001-backup-format.md:30]: Please remove this

File: docs/rfc/001-backup-format.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
+# Structured Backup format
+
+## Principles
+
+- Based-zip64 format (compression level: archive)
+- Based-zip64 standard file encryption
+- Structured tree structure
```
Comment (Jack-Works) [inline docs/rfc/001-backup-format.md:7]: Should we have an integrity mechanism?

File: docs/rfc/001-backup-format.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+```plain
+README.txt
+
+# core file integrity check
+files.json # file hash list with manifest* and README.txt
+
+# the backup metadata
+manifest.json
+  - version = 3?
```
Comment (Jack-Works) [inline docs/rfc/001-backup-format.md:19]: We can start with 0 or 1 because it is a binary format.

File: docs/rfc/001-backup-format.md (hunk 4 above)
Comment (septs) [reply docs/rfc/001-backup-format.md:30]: youself update this file?

File: docs/rfc/001-backup-format.md (hunk 6 above)
Comment (septs) [reply docs/rfc/001-backup-format.md:19]: approve, i think you self update the document.

**Comment by Jack-Works** [comment]
I have converted this PR into a draft. Because I think it is still missing many details. @septs we should complete this with each file's format (in TS or other format).

File: docs/rfc/001-backup-format.md (hunk 5 above)
Comment (septs) [reply docs/rfc/001-backup-format.md:7]: generate all `manifest-*.signed` file hash to one result (sha2-512, sha3-512) to remote logging

```
if query the logging exists then
   verified
else
   unverified
```

**Comment by septs** [comment]
@Jack-Works structured all json file with json schema?

**Comment by Jack-Works** [comment]
> @Jack-Works structured all json file with json schema?

We can start from TypeScript definition

**Pull request closed without merging**